File: eureka/ios/nativeLib/nativeLib/Database/dbTables/UserTable.swift
//
//  UserTable.swift
//  LifePlus
//
//  Created by work on 10/09/20.
//

import Foundation

public class UserTable {
    public static let TABLE_NAME: String = "users"

    public class Cols {
      public static let ID: String = "id"
      public static let NAME: String = "name"
      public static let BIRTH_DATE: String = "birth_date"
      public static let GENDER_ID: String = "gender_id"
      public static let ETHNICITY_ID: String = "ethnicity_id"
      public static let SKIN_TONE_ID: String = "skin_tone_id"
      public static let ADDRESS: String = "address"

      public static let COUNTRY: String = "country"
      public static let ZIP: String = "zip"
      public static let PASSWORD: String = "password"
      public static let HEIGHT_FT: String = "height_ft"
      public static let HEIGHT_IN: String = "height_in"
      public static let WEIGHT: String = "weight"

      public static let WEIGHT_UNIT: String = "weight_unit"
      public static let TNC_DATE: String = "tnc_date"
      public static let STEP_GOAL: String = "step_goal"
      public static let HW_ID: String = "hw_id"
      public static let GLUCOSE_UNIT: String = "glucose_unit"
      public static let AUTO_MEASURE: String = "auto_measure"

      public static let AUTO_FREQUENCY: String = "auto_frequency"
      public static let SLEEP_TRACKING: String = "sleep_tracking"
      public static let POWER_SAVE: String = "power_save"
      public static let CGM_DEBUG: String = "cgm_debug"
      public static let REGISTRATION_STATE: String = "registration_state"
      public static let WEATHER_UNIT: String = "weather_unit"
      public static let UPDATE_DATE: String = "update_date"
      public static let UPLOAD_DATE: String = "upload_date"
        
      // calculated field
      public static let AGE: String = "age"
      public static let AGE_EXP: String = "cast(ifnull(strftime('%Y.%m%d', 'now') - strftime('%Y.%m%d', \(BIRTH_DATE)), 0) as int) AS \(AGE)"
    }
}

File: eureka/android/library/src/main/java/com/lifeplus/Database/dbTables/UserTable.java
package com.lifeplus.Database.dbTables;

public class UserTable {
    public static final String TABLE_NAME = "users";

    public static class Cols {
        public static final String ID = "id";
        public static final String NAME = "name";
        public static final String BIRTH_DATE = "birth_date";
        public static final String GENDER_ID = "gender_id";
        public static final String ETHNICITY_ID = "ethnicity_id";
        public static final String SKIN_TONE_ID = "skin_tone_id";
        public static final String ADDRESS = "address";

        public static final String COUNTRY = "country";
        public static final String ZIP = "zip";
        public static final String PASSWORD = "password";
        public static final String HEIGHT_FT = "height_ft";
        public static final String HEIGHT_IN = "height_in";
        public static final String WEIGHT = "weight";

        public static final String WEIGHT_UNIT = "weight_unit";
        public static final String TNC_DATE = "tnc_date";
        public static final String STEP_GOAL = "step_goal";
        public static final String HW_ID = "hw_id";
        public static final String GLUCOSE_UNIT = "glucose_unit";
        public static final String AUTO_MEASURE = "auto_measure";

        public static final String AUTO_FREQUENCY = "auto_frequency";
        public static final String SLEEP_TRACKING = "sleep_tracking";
        public static final String POWER_SAVE = "power_save";
        public static final String CGM_DEBUG = "cgm_debug";
        public static final String REGISTRATION_STATE = "registration_state";
        public static final String WEATHER_UNIT = "weather_unit";
        public static final String UPDATE_DATE = "update_date";
        public static final String UPLOAD_DATE = "upload_date";

        // calculated field
        public static final String AGE = "age";
        public static final String AGE_EXP = "cast(ifnull(strftime('%Y.%m%d', 'now') - strftime('%Y.%m%d', " + BIRTH_DATE + "), 0) as int) AS " + AGE;
    }
}
